### PR TITLE
feat: implement TS compressor methods

### PR DESCRIPTION
**Pull request by taylorsw**
Draft PR.

### CODE REVIEW

File: typescript-id-allocator/types/persisted-types/0.0.1.ts [hunk 1]
```diff
@@ -105,14 +12,7 @@ export interface VersionedSerializedIdCompressor {
  * The serialized contents of an IdCompressor, suitable for persistence in a summary.
  */
 export interface SerializedIdCompressor extends VersionedSerializedIdCompressor {
-	/** The cluster capacity of this compressor */
```
Comment (taylorsw) [inline typescript-id-allocator/types/persisted-types/0.0.1.ts:108]: This file: just deleting the old persisted types + adding bytes here.

File: typescript-id-allocator/util/idRange.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// COPIED FROM FLUID FRAMEWORK
```
Comment (taylorsw) [inline typescript-id-allocator/util/idRange.ts:1]: @crichards17 don't review this, it's just copied and pasted

File: typescript-id-allocator/types/persisted-types/0.0.1.ts [hunk 3]
```diff
@@ -105,14 +12,7 @@ export interface VersionedSerializedIdCompressor {
...
-	readonly clusterCapacity: number;
-	/** The number of reserved IDs in this compressor */
-	readonly reservedIdCount: number;
-	/** All sessions except the local session. */
-	readonly sessions: readonly SerializedSessionData[];
-	/** All clusters in the compressor in the order they were created. */
-	readonly clusters: readonly SerializedCluster[];
+	readonly bytes: Uint8Array;
```
Comment (taylorsw) [inline typescript-id-allocator/types/persisted-types/0.0.1.ts:15]: This file: just deleting the old persisted types + adding bytes here.

File: rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs [hunk 4]
```diff
@@ -63,14 +63,14 @@ impl IdCompressor {
     pub fn take_next_range(&mut self) -> InteropIdRange {
         let token = self.compressor.get_local_session_token() as f64;
         match self.compressor.take_next_range().range {
-            Some((local, count)) => InteropIdRange {
+            Some((first_local, count)) => InteropIdRange {
```
Comment (taylorsw) [inline rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs:66]: this file just contains renames

File: typescript-id-allocator/IdCompressor.ts [hunk 5]
```diff
@@ -84,4 +132,21 @@ export class IdCompressor implements IIdCompressor, IIdCompressorCore {
 	public serialize(withSession: boolean): SerializedIdCompressor {
 		throw new Error("Method not implemented.");
 	}
+
+	public static deserialize(serialized: SerializedIdCompressorWithOngoingSession): IdCompressor;
```
Comment (taylorsw) [inline typescript-id-allocator/IdCompressor.ts:136]: Just copied the same signature from FF, leaving unimplemented for now

**Review by crichards17 — APPROVED** [review]

**Pull request merged**